stl/inc/vector

```diff
@@ -579,24 +494,37 @@ private:
             return;
         }
 
-        const auto _First    = reinterpret_cast<const char*>(_Unfancy(_First_));
-        const auto _End      = reinterpret_cast<const char*>(_Unfancy(_End_));
-        const auto _Old_last = reinterpret_cast<const char*>(_Unfancy(_Old_last_));
-        const auto _New_last = reinterpret_cast<const char*>(_Unfancy(_New_last_));
-        if constexpr (_Has_minimum_allocation_alignment<vector>) {
-            __sanitizer_annotate_contiguous_container(_First, _End, _Old_last, _New_last);
+        const void* const _First    = _STD _Unfancy(_First_);
+        const void* const _End      = _STD _Unfancy(_End_);
+        const void* const _Old_last = _STD _Unfancy(_Old_last_);
+        const void* const _New_last = _STD _Unfancy(_New_last_);
+        if constexpr ((_Container_allocation_minimum_alignment<vector>) > _Asan_granularity) {

```

Comment:
Same in string.

Reply:
Technically no, since ASan will just push `_End` to the end of the shadow memory section anyways; however, it may make sense to do so anyways in case this function starts _actually_ checking.

Reply:
I see - both this check and the `_Small/Large_string_always_asan_aligned` checks in string are just optimizations then, yeah? It makes the code (at least, when you don't see how it all fits together) quite a bit more complex - might be worth removing the optimization?

Reply:
Ah, because we still want to over-poison when using the default allocator to catch right oob situations when the capacity doesn't line up? Might want to add comments about that being the strategy, perhaps to the xmemory comment?